python/src/gluon_ir.cc

```diff
@@ -181,7 +185,20 @@ py::object layoutToGluon(Attribute layout) {
         toStdVector(ctaLayout.getCTAOrder()));
   } else if (auto autoEnc = dyn_cast<gluon::AutoEncodingAttr>(layout)) {
     return layouts.AutoLayout();
+  } else if (auto amdMfma = dyn_cast<ttg::AMDMfmaEncodingAttr>(layout)) {
+    auto elemType = amdMfma.getElementType();
+    bool isFp32 =
+        !elemType.has_value() || isa<mlir::Float32Type>(elemType.value());
+    assert(isFp32 && "Only float32 is supported for now in AMF MFMA encoding");
```

Comment:
Any reason we cannot fp64 also? At least maybe `throw py::value_error("AMD MFMA layout only supports fp32 element type for now")` to give a better error message.